workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourLibrary.cs

```diff
@@ -0,0 +1,198 @@
+            {
+                // Figure out if reader or writer
+                // Get component ID
+                // Store in data structures
+                Type requiredType = adapter.TypeOfMember;
+                var isReader = Attribute.IsDefined(requiredType, typeof(ReaderInterfaceAttribute), false);
+                var isWriter = Attribute.IsDefined(requiredType, typeof(WriterInterfaceAttribute), false);
+                if (!isReader && !isWriter)
+                {
+                    logger.HandleLog(LogType.Warning, new LogEvent(BadRequiredMemberWarning)
+                        .WithField(LoggingUtils.LoggerName, LoggerName)
+                        .WithField("MonoBehaviour", behaviourType.Name)
+                        .WithField("RequiredType", requiredType.Name));
+                    continue;
+                }
+
+                var componentIdAttribute =
+                    (ComponentIdAttribute) Attribute.GetCustomAttribute(requiredType, typeof(ComponentIdAttribute), false);
+                if (componentIdAttribute == null)
+                {
+                    logger.HandleLog(LogType.Error, new LogEvent(MalformedReaderOrWriter)
+                        .WithField(LoggingUtils.LoggerName, LoggerName)
+                        .WithField("MonoBehaviour", behaviourType.Name)
+                        .WithField("RequiredType", requiredType.Name));
+                    continue;
+                }
+
+                var componentId = componentIdAttribute.Id;
+                if (componentIdsToAdapters.ContainsKey(componentId))
+                {
+                    logger.HandleLog(LogType.Error, new LogEvent(MultipleReadersWritersRequiredError)
+                        .WithField(LoggingUtils.LoggerName, LoggerName)
+                        .WithField("MonoBehaviour", behaviourType.Name)
+                        .WithField("ComponentID", componentId)
+                        .WithField("RequiredType", requiredType.Name));
+                    invalidMonoBehaviourTypes.Add(behaviourType);
+                    break;
+                }
+
+                componentIdsToAdapters[componentId] = adapter;
+                if (isReader)
+                {
+                    readerIds.Add(componentId);
+                }
+                else
+                {
+                    writerIds.Add(componentId);
+                }
+            }
+        }
+
+        private const BindingFlags FieldFlags = BindingFlags.Instance | BindingFlags.NonPublic |
+            BindingFlags.Public | BindingFlags.DeclaredOnly;
+
+        private const BindingFlags PropertyFlags =
+            FieldFlags | BindingFlags.SetProperty | BindingFlags.GetProperty;
+
+        private List<IMemberAdapter> GetMembersWithMatchingAttributes(Type targetType)
+        {
+            List<IMemberAdapter> adapters = new List<IMemberAdapter>();
+            foreach (var property in targetType.GetProperties(PropertyFlags))
```

Comment:
I think a quick survey of AAA, Spilt Milk and Klang about whether or not they actually inject into Properties is in order. If they don't, we can drop all of this extra machinery, interfaces and classes that try to normalize field/property handling.

Reply:
What should be the process with this PR for the while? It's kind of annoying if merging it has to wait for that, when we are trying to complete the monobehaviour feature set. Maybe insist on having an answer before we merge the feature branch into master?

Reply:
I think leave it out and add it in later if we need it. It's not something I've ever seen used and seems brittle.